### PR TITLE
[veomni] feat: Add GRPO training scripts for Qwen3-VL-30B-MOE (VeOmni Backends)

### PR DESCRIPTION
**Pull request by phdddd**
### What does this PR do?

Add NPU/GPU GRPO training scripts for Qwen3-VL-30B-MOE (VeOmni backends).

### Checklist Before Starting

- [ ] Search for similar PRs. Paste at least one query link here: ...
- [ ] Format the PR title as `[{modules}] {type}: {description}` (This will be checked by the CI)
  - `{modules}` include `fsdp`, `megatron`, `veomni`, `sglang`, `vllm`, `rollout`, `trainer`, `ci`, `training_utils`, `recipe`, `hardware`, `deployment`, `ray`, `worker`, `single_controller`, `misc`, `perf`, `model`, `algo`, `env`, `tool`, `ckpt`, `doc`, `data`, `cfg`, `reward`
  - If this PR involves multiple modules, separate them with `,` like `[megatron, fsdp, doc]`
  - `{type}` is in `feat`, `fix`, `refactor`, `chore`, `test`
  - If this PR breaks any API (CLI arguments, config, function signature, etc.), add `[BREAKING]` to the beginning of the title.
  - Example: `[BREAKING][fsdp, megatron] feat: dynamic batching`

### Test

The relevant testing of this script is in progress and will be added here soon.

### API and Usage Example

> Demonstrate how the API changes if any, and provide usage example(s) if possible.

```python
# Add code snippet or script demonstrating how to use this
```

### Design & Code Changes

> Demonstrate the high-level design if this PR is complex, and list the specific changes.

### Checklist Before Submitting

> [!IMPORTANT]
> Please check all the following items before requesting a review, otherwise the reviewer might deprioritize this PR for review.

- [ ] Read the [Contribute Guide](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md).
- [ ] Apply [pre-commit checks](https://github.com/volcengine/verl/blob/main/CONTRIBUTING.md#code-linting-and-formatting): `pre-commit install && pre-commit run --all-files --show-diff-on-failure --color=always`
- [ ] Add / Update [the documentation](https://github.com/volcengine/verl/tree/main/docs).
- [ ] Add unit or end-to-end test(s) to [the CI workflow](https://github.com/volcengine/verl/tree/main/.github/workflows) to cover all the code. If not feasible, explain why: ...
- [ ] Once your PR is ready for CI, send a message in [the `ci-request` channel](https://verl-project.slack.com/archives/C091TCESWB1) in [the `verl` Slack workspace](https://join.slack.com/t/verl-project/shared_invite/zt-3855yhg8g-CTkqXu~hKojPCmo7k_yXTQ). (If not accessible, please try [the Feishu group (飞书群)](https://applink.larkoffice.com/client/chat/chatter/add_by_link?link_token=772jd4f1-cd91-441e-a820-498c6614126a).)
- [ ] If your PR is related to the `recipe` submodule, please also update the reference to the submodule commit via `git submodule update --remote` or `cd recipe && git pull origin main`.

### CODE REVIEW

**Review by ji-huazhong — COMMENTED** [review]
Thanks for your contribution. Could you please provide the corresponding experiment report?

File: examples/sft/vlm/run_qwen3_vl_30b_moe_veomni.sh [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    optim.clip_grad=1.0 \
+    optim.lr_min=1e-6 \
+    optim.lr_scheduler_type=cosine \
+    engine.ulysses_parallel_size=${SP_SIZE} \
+    engine.fsdp_size=${FSDP_SIZE} \
+    engine.expert_parallel_size=${EP_SIZE} \
+    engine.enable_full_shard=True \
+    engine.moe_implementation=fused \
+    engine.attn_implementation=flash_attention_2 \
+    engine.param_ofload=True \
```
Comment (deerlu) [inline examples/sft/vlm/run_qwen3_vl_30b_moe_veomni.sh]: 这个是拼写错误？

**Review by wucong25 — DISMISSED** [review]

File: examples/grpo_trainer/run_qwen3_vl_30b_moe_veomni.sh [hunk 2]
```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env bash
+# This script is a demo for GRPO training of Qwen3-VL-30B-MOE using VeOmniEngine.
```
Comment (wucong25) [inline examples/grpo_trainer/run_qwen3_vl_30b_moe_veomni.sh]: 刷新一下grpo_trainer下的readme和模型支持列表

Comment (wucong25) [inline examples/grpo_trainer/run_qwen3_vl_30b_moe_veomni.sh]: GRPO | Qwen2.5-32B | FSDP training | NVIDIA GPUs or Ascend NPUs  需要在脚本最上面注明NPU也支持

File: examples/grpo_trainer/run_qwen3_vl_30b_moe_veomni.sh [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+
+set -xeuo pipefail
+
+# ---- user-adjustable ----
+data_path=${data_path:-$HOME/geo3k}
+model_path=${model_path:-$HOME/Qwen3-VL-30B-A3B-Instruct}
+
+nnodes=${nnodes:-2}
+num_gpus_per_node=8
```
Comment (wucong25) [inline examples/grpo_trainer/run_qwen3_vl_30b_moe_veomni.sh:14]: 注意到num_gpus_per_node=8，npu上是用A2验证的么

File: examples/grpo_trainer/run_qwen3_vl_30b_moe_veomni.sh (hunk 2 above)
Comment (phdddd) [reply examples/grpo_trainer/run_qwen3_vl_30b_moe_veomni.sh]: done

Comment (phdddd) [reply examples/grpo_trainer/run_qwen3_vl_30b_moe_veomni.sh]: done

File: examples/grpo_trainer/run_qwen3_vl_30b_moe_veomni.sh (hunk 3 above)
Comment (phdddd) [reply examples/grpo_trainer/run_qwen3_vl_30b_moe_veomni.sh:14]: npu是nnodes=${nnodes:-1}
num_gpus_per_node=16
这里按照GPU的写法是否可以，其他部分都是通用的

**Review by wucong25 — APPROVED** [review]
